### PR TITLE
UPI vSphere platform:none workflow

### PR DESCRIPTION
**Pull request by mansikulkarni96**
This PR updates the command script [vsphere-upi-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh) to use platform:none as the platform type in the install-config file.
It also renames and modifies chain upi-conf-vsphere to upi-conf-vsphere-ovn which is used in the UPI vSphere workflow by adding the steps required for setting up hybrid OVN networking required for usage in Windows e2e tests. 
It also updates the WMCO config to add a test that runs the UPI vSphere platform:none workflow and updates the jobs to reflect the same.

Ran command:
`make update`

### CODE REVIEW

**Comment by mansikulkarni96** [comment]
ci-openshift-windows-machine-config-operator-release-4.9-platform-none-e2e-operator is failing in step `ipi-conf-vsphere-check` when making a [vsphere_datacenter call](https://github.com/openshift/release/blob/ef4ae26d45cdd6a2633016c07fbf2fe86c0841f7/ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh#L78).

File: ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.9-presubmits.yaml [hunk 1]
```diff
@@ -412,6 +412,79 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )lint,?($|\s.*)
+  - agent: kubernetes
+    always_run: false
+    branches:
+    - ^release-4\.9$
+    - ^release-4\.9-
+    cluster: build02
```
Comment (rvanderp3) [inline ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.9-presubmits.yaml:420]: @mansikulkarni96 I think this should be vsphere and not build02.  This would possibly explain why neither the VMC or IBM vCenters are reachable.

File: ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh [hunk 2]
```diff
@@ -102,15 +102,7 @@ compute:
 - name: "worker"
   replicas: 0
 platform:
-  vsphere:
-    vcenter: "${vsphere_url}"
-    datacenter: "${vsphere_datacenter}"
```
Comment (mansikulkarni96) [inline ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh:107]: Not a 100% sure but it looks like not having the datacenter set, is causing the gather-extra [artifacts](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/22371/rehearse-22371-pull-ci-openshift-windows-machine-config-operator-release-4.9-platform-none-e2e-operator/1445780737910378496/artifacts/platform-none-e2e-operator/gather-extra/build-log.txt) not being collected.

**Review by rvanderp3 — COMMENTED** [review]
The naming convention of the step `upi-conf-vsphere-ovn` implies to me that this would be a typical vSphere install with the cloud provider enabled.  Since this chain is specifically creating a non-standard ovn installation of platform none on vmware, can we rename the step to something like: `upi-conf-vsphere-platform-none-ovn`.  There is the possibility that we could create a UPI OVN step for a traditional vSphere installation in the future.

File: ci-operator/step-registry/upi/vsphere/pre/upi-vsphere-pre-chain.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ chain:
   as: upi-vsphere-pre
   steps:
   - ref: ipi-install-rbac
-  - chain: upi-conf-vsphere
+  - chain: upi-conf-vsphere-platform-none-ovn
```
[the remark this answers is not in the document]
Comment (mansikulkarni96) [reply ci-operator/step-registry/upi/vsphere/pre/upi-vsphere-pre-chain.yaml]: I have updated the PR to reflect what we discussed, the only confusing bit to me atm is since I am updating the current [ upi-conf-vsphere-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh) to remove platform specific information,  doesn't it technically also make the existing [upi-conf-vsphere](https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-chain.yaml) chain which is using it to be included under the platform-none umbrella/ directory?

File: ci-operator/step-registry/upi/vsphere/platform-none/ovn/pre/upi-vsphere-platform-none-ovn-pre-chain.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+chain:
+  as: upi-vsphere-platform-none-ovn-pre
+  steps:
+  - ref: ipi-install-rbac
+  - chain: upi-conf-vsphere-platform-none-ovn
```
Comment (mansikulkarni96) [inline ci-operator/step-registry/upi/vsphere/platform-none/ovn/pre/upi-vsphere-platform-none-ovn-pre-chain.yaml:5]: `upi-conf-vsphere-platform-none-ovn` is the chain used currently in platform-none pre-chain.

File: ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh [hunk 5]
```diff
@@ -102,15 +102,7 @@ compute:
 - name: "worker"
   replicas: 0
 platform:
-  vsphere:
```
[the remark this answers is not in the document]
Comment (mansikulkarni96) [reply ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh]: I had to add another test step that will be used by both the vsphere and platform-none workflows to create manifests an other supporting files as the existing upi-vsphere-command.sh script did the creating of install config as well as consumed it in the same script.

**Comment by mansikulkarni96** [comment]
/test  pj-rehearse 
The submodule clone issue has been fixed in master/4.10

**Comment by mansikulkarni96** [comment]
@rvanderp3 I still see the same bootstrapping issue on 4.10/master jobs, 
`level=debug msg="Still waiting for the Kubernetes API: Get \"https://api.ci-op-m3lq6vd6-bce97.vmc-ci.devcluster.openshift.com:6443/version?timeout=32s\": EOF"
level=error msg="Attempted to gather ClusterOperator status after wait failure: listing ClusterOperator objects: Get \"https://api.ci-op-m3lq6vd6-bce97.vmc-ci.devcluster.openshift.com:6443/apis/config.openshift.io/v1/clusteroperators\": EOF"
level=error msg="Bootstrap failed to complete: Get \"https://api.ci-op-m3lq6vd6-bce97.vmc-ci.devcluster.openshift.com:6443/version?timeout=32s\": EOF"
level=error msg="Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane."
level=fatal msg="Bootstrap failed to complete"`

**Comment by mansikulkarni96** [comment]
Not sure why the 4.8 job is landing on okd.
On 4.9, 4.10 the issue seems related to the vsphere-csi-driver session
/test pj-rehearse

**Comment by mansikulkarni96** [comment]
Cluster came up successfully on all versions with the latest experiment of changing platform type in a previous step and parsing the install-config to make the decision on adding vSphere platform type specific details.

**Comment by mansikulkarni96** [comment]
Changed the job config of 4.10 rehearsal to point to the upi-vsphere workflow, to show the difference between install-configs:
platform none case: [log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/22371/rehearse-22371-pull-ci-openshift-windows-machine-config-operator-release-4.9-platform-none-vsphere-e2e-operator/1458084088320626688/artifacts/platform-none-vsphere-e2e-operator/upi-conf-vsphere/build-log.txt)
platform vsphere case: [log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/22371/rehearse-22371-pull-ci-openshift-windows-machine-config-operator-release-4.10-platform-vsphere-e2e-operator/1458084088240934912/artifacts/platform-vsphere-e2e-operator/upi-conf-vsphere/build-log.txt)

**Review by aravindhp — APPROVED** [review]
/lgtm

for WMCO
